### PR TITLE
Add profile version to archive

### LINKED ISSUE #1350: inspec-archive should create a zip with version string
### Description

When packaging up a profile using `inspec archive` the tarball/zip that gets created is simple `<profilename.tar.gz>`

This makes it difficult to do custom downloads without chef compliance server and other distribution methods, as versioning them becomes difficult.

### InSpec and Platform Version

`inspec 1.7.1`
`train 0.22.0`

### Replication Case

Simple create a profile and run `inspec archive <path>`

### Possible Solutions

It seems reasonably easy to have the created tarball be `<profilename>-<version>.tar.gz`

### Stacktrace

```
inspec archive inspec-pcs --diagnose                                                                                                                                                   
InSpec version: 1.7.1
Train version: 0.22.0
Command line configuration:
{"zip"=>false,
 "tar"=>false,
 "overwrite"=>false,
 "ignore_errors"=>false,
 "diagnose"=>true}
JSON configuration file:
{}
Merged configuration:
{"zip"=>false,
 "tar"=>false,
 "overwrite"=>false,
 "ignore_errors"=>false,
 "diagnose"=>true}

I, [2016-12-08T14:17:06.725656 #67115]  INFO -- : Checking profile in inspec-pcs
I, [2016-12-08T14:17:06.725725 #67115]  INFO -- : Metadata OK.
I, [2016-12-08T14:17:06.787953 #67115]  INFO -- : Found 5 controls.
W, [2016-12-08T14:17:06.788011 #67115]  WARN -- : Control pcs cluster status has no description
W, [2016-12-08T14:17:06.788040 #67115]  WARN -- : Control pcs links has no description
W, [2016-12-08T14:17:06.788053 #67115]  WARN -- : Control PCS License Status has no description
W, [2016-12-08T14:17:06.788063 #67115]  WARN -- : Control pcs mounts has no description
W, [2016-12-08T14:17:06.788074 #67115]  WARN -- : Control pcs services has no description
I, [2016-12-08T14:17:06.788272 #67115]  INFO -- : Archive /Users/Lee/git/pcs.tar.gz exists already. Use --overwrite.
```

### PR DESCRIPTION
**Pull request by jaxxstorm**
Fixes #1350 

Probably needs some discussion here...

### CODE REVIEW

File: lib/inspec/profile.rb [hunk 1]
```diff
@@ -399,9 +399,12 @@ def archive_name(opts)
       name = params[:name] ||
              fail('Cannot create an archive without a profile name! Please '\
                   'specify the name in metadata or use --output to create the archive.')
+      version = params[:version] ||
+             fail('Cannot create an archive with a profile version! Please '\
+                  'specify the version in metadat or use --output to create the archive.')
```
Comment (chris-rock) [inline lib/inspec/profile.rb]: `metadata`

**Comment by chris-rock** [comment]
Great improvement @jaxxstorm Can you rebase and fix the lint error? Happy to merge it then.

**Comment by jaxxstorm** [comment]
@chris-rock I fixed the lint error, but we're still failing here. Any tips?

**Comment by chris-rock** [comment]
@jaxxstorm We need to update our functional tests as well:
https://github.com/chef/inspec/blob/master/test/functional/inspec_archive_test.rb#L12-L15
and https://github.com/chef/inspec/blob/master/test/functional/inspec_archive_test.rb#L28-L31

You can quickly run the specific tests locally with: `bundle exec ruby -W -Ilib:test test/functional/inspec_archive_test.rb`

**Comment by jaxxstorm** [comment]
Woot, finally got there, thanks @chris-rock !

**Comment by chris-rock** [comment]
@jaxxstorm Cool. Thank you for the quick turnaround!

**Pull request merged**